sources/api/schnauzer/tests/multicall_integration.rs

```diff
@@ -0,0 +1,230 @@
+    let stderr_v2 = String::from_utf8_lossy(&output_v2.stderr);
+    assert_ne!(
+        stderr_v1, stderr_v2,
+        "v1 and v2 should produce different outputs"
+    );
+}
+
+/// Test error handling for unrecognized program names
+#[test]
+fn test_unrecognized_program_name_defaults_to_v2() {
+    // Given A multicall binary with an unrecognized symlink name
+    let test = MulticallTest::new();
+    let unknown_link = test.create_symlink("unknown-tool");
+
+    // When Invoking the binary with the unknown name
+    let output = test.run_command(&unknown_link, &["--help"]);
+
+    // Then Should default to v2 behavior
+    assert!(
+        test.is_v2_output(&output),
+        "Unknown tool name should default to v2"
+    );
+}
+
+/// Test that both versions handle basic argument parsing
+#[test]
+fn test_argument_handling_preservation() {
+    // Given A multicall binary with symlinks for both tools
+    let test = MulticallTest::new();
+    let schnauzer_link = test.create_symlink("schnauzer");
+    let schnauzer_v2_link = test.create_symlink("schnauzer-v2");
+
+    // When Invoking each with invalid arguments
+    let output_v1 = test.run_command(&schnauzer_link, &["invalid-arg"]);
+    let output_v2 = test.run_command(&schnauzer_v2_link, &["invalid-command"]);
+
+    // Then Both should handle arguments and produce different error messages
+    assert!(output_v1.status.code().is_some());
+    assert!(output_v2.status.code().is_some());
+
+    // Verify they maintain their distinct error handling
+    let stderr_v1 = String::from_utf8_lossy(&output_v1.stderr);
+    let stderr_v2 = String::from_utf8_lossy(&output_v2.stderr);
+    assert_ne!(stderr_v1, stderr_v2, "Error messages should be different");
+}
+
+/// Test that the multicall binary preserves exit codes
+#[test]
+fn test_exit_code_preservation() {
+    // Given A multicall binary with a schnauzer symlink
+    let test = MulticallTest::new();
+    let schnauzer_link = test.create_symlink("schnauzer");
+
+    // When Invoking the symlink with --help
+    let output = test.run_command(&schnauzer_link, &["--help"]);
+
+    // Then Should preserve the exit code from the underlying implementation
+    assert_eq!(output.status.code(), Some(USAGE_ERROR_EXIT_CODE));
+}
+
+/// Integration test to verify the multicall mechanism works end-to-end
+#[test]
+fn test_multicall_integration() {
+    // Given A multicall binary with symlinks for both tools
+    let test = MulticallTest::new();
+    let schnauzer_link = test.create_symlink("schnauzer");
+    let schnauzer_v2_link = test.create_symlink("schnauzer-v2");
+
+    // When Executing both symlinks with --help
+    let output_v1 = test.run_command(&schnauzer_link, &["--help"]);
+    let output_v2 = test.run_command(&schnauzer_v2_link, &["--help"]);
+
+    // Then Both should execute and produce different outputs
+    assert!(output_v1.status.code().is_some());
+    assert!(output_v2.status.code().is_some());
```

Comment:
This isn't a very useful check; the status code will always be present unless the process gets killed by a signal.